public/app/features/explore/Explore.tsx

```diff
@@ -268,11 +269,11 @@ export class Explore extends React.PureComponent<ExploreProps, ExploreState> {
     });
   };
 
-  refreshExplore = () => {
-    const { exploreId, update } = this.props;
+  refreshExplore = (prevUrlQuery: string) => {
+    const { exploreId, urlQuery } = this.props;
 
-    if (update.queries || update.range || update.datasource || update.mode) {
-      this.props.refreshExplore(exploreId);
+    if (urlQuery !== prevUrlQuery && urlQuery !== lastSavedUrl[exploreId]) {
+      this.props.refreshExplore(exploreId, urlQuery);
```

Comment:
Before we already had the diff in the state. Instead we just moved the diffing inside the `refreshExplore` so the diff is computed when needed.